recipe/gen_patch_json.py

```diff
@@ -1830,6 +1830,22 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
                     new_deps.append(dep)
             record["depends"] = new_deps
 
+        # ipykernel >=4.0.1,<6.5.0 needs ipython_genutils. Old versions of
+        # ipython and traitlets depend on ipython_genutils so the dependency
+        # was originally satisfied indirectly. Newer versions of ipython and
+        # traitlets don't pull in ipython_genutils anymore so we need to make
+        # that dependency explicit.
+        if (record_name == "ipykernel" and
```

Comment:
Can you add a timestamp check here as well? Thanks!
```suggestion
        if (record_name == "ipykernel" and record.get("timestamp", 0) <= 1664184744000 and
```